src/main/java/reposense/parser/AuthorConfigLocationParser.java

```diff
@@ -0,0 +1,36 @@
+package reposense.parser;
+
+import static reposense.parser.LocalRepoLocationParser.parseLocalRepoLocation;
+import static reposense.parser.RemoteRepoLocationParser.parseRemoteRepoLocation;
+import static reposense.util.SystemUtil.isValidUrl;
+
+import java.util.List;
+
+/**
+ * Contains repository location parsing functionalities specified to {@code author-config.csv} file.
+ */
+public class AuthorConfigLocationParser {
+    public static final String LOCATION_BRANCHES_DELIMITER = "~";
+    public static final String BRANCHES_DELIMITER = "|";
+    public static final String SPLIT_SAFE_BRANCHES_DELIMITER = "\\" + BRANCHES_DELIMITER;
+
+    /**
+     * Parse a location and returns a list containing location and branch information
+     * according to whether it uses no additional special syntax, delimiter syntax or
+     * GitHub-specific branch URL syntax.
+     * The first item in the list is the location, while subsequent items are the
+     * branches specified.
+     */
+    public static List<String> parseLocation(String location, String defaultSpecifiedBranch) {
+        if (isRemoteRepo(location)) {
+            return parseRemoteRepoLocation(location, defaultSpecifiedBranch);
+        } else {
+            return parseLocalRepoLocation(location, defaultSpecifiedBranch);
+        }
+    }
+
+    private static boolean isRemoteRepo(String location) {
```

Comment:
Does this actually work? I don't think `|` is a valid URL character so a remote repo with multiple branches specified fails this unless I'm understanding wrongly. It also ignores remote repos specified by the SSH protocol. I think we should try a more agnostic implementation that does not care whether it is remote or local

Reply:
https://stackoverflow.com/a/3931701 Following this discussion and the comments below and Java 8's own URL API, I don't actually think the validation does all that much besides checking if a protocol is specified

Reply:
Thank you for spotting this! You are right, the code seems to be defaulting the remote repos into the local repos in this case. Given that both local and remote can be handled the same way, I have taken your advice to remove the separation in the most recent commit.